### PR TITLE
BaSP-TG-56: Amadeo-Muises-Table Component

### PR DESCRIPTION
**Pull request by RamiroAmadeo**
Make table a shared component

### CODE REVIEW

File: src/Components/Admins/Modal/modal.module.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+.header__button {
+    padding: 0;
+    border: none;
+    background-color: white;
+    color: var(--main-color);
+    cursor: pointer;
+}
+
+/* :root {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:96]: Remove this comment

File: src/Components/Admins/Modal/modal.module.css [hunk 2]
```diff
@@ -0,0 +1,100 @@
+.modal__display__none {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:1]:
```suggestion
.modal-display-none {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+    width: 90%;
+    margin-top: 10px;
+    border-bottom: 1px solid rgba(0, 0, 0, 0.132);
+    padding-bottom: 10px;
+}
+
+.header__title {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:43]:
```suggestion
.header-title {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    gap: 1.2em;
+    height: 100%;
+    width: 85%;
+    margin-block: 20px;
+}
+
+.content__message {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:58]:
```suggestion
.content-message {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    max-width: 100%;
+    font-size: 15px;
+    white-space: pre-line;
+}
+
+.options {
+    align-self: flex-end;
+}
+
+.options__button {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:69]:
```suggestion
.options-button {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    margin: 0 10px;
+    padding: 10px 28px;
+    border: none;
+    border-radius: 18px;
+    color: white;
+    cursor: pointer;
+    box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.25);
+}
+
+.options__close {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:79]:
```suggestion
.options-close {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    background-color: var(--close-color)
+}
+
+.options__close:hover {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:83]:
```suggestion
.options-close:hover {
```

File: src/Components/Admins/Modal/modal.module.css [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    transition: 0.2s;
+    background-color: var(--hover-close-color)
+}
+
+.header__button {
```
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:88]:
```suggestion
.header-button {
```

File: src/Components/Shared/Table/index.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import tableStyles from './index.module.css';
+import Logo from '../Assets/logoDelete.png';
+import { useHistory } from 'react-router-dom';
+
+const Table = ({ data, headers, urlForm, showModal, deleteId }) => {
+  console.log(data);
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:7]: Remove this console log

File: src/Components/Shared/Table/index.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import tableStyles from './index.module.css';
+import Logo from '../Assets/logoDelete.png';
+import { useHistory } from 'react-router-dom';
+
+const Table = ({ data, headers, urlForm, showModal, deleteId }) => {
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:6]: We should receive two functions as props: onDelete and onRowClick

Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:6]:
```suggestion
const Table = ({ data, headers, onDelete, onRowClick }) => {
```

File: src/Components/Shared/Table/index.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+          </tr>
+        </thead>
+        <tbody className={tableStyles.tbody}>
+          {data.map((row) => {
+            return (
+              <>
+                <tr
+                  className={tableStyles.tr}
+                  key={row._id}
+                  onClick={() => history.push(`${urlForm}${row._id}`)}
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:27]:
```suggestion
                  onClick={onRowClick}
```

File: src/Components/Shared/Table/index.js [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+                      <>
+                        <td key={index}>{row[header]}</td>
+                      </>
+                    );
+                  })}
+                  <td>
+                    <img
+                      className={tableStyles.logo}
+                      src={Logo}
+                      onClick={(e) => {
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:40]:
```suggestion
                      onClick={(e) => {
                        e.stopPropagation();
                        onDelete();
```

File: src/Components/Shared/Table/index.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+                      }}
+                    />
+                  </td>
+                </tr>
+              </>
+            );
+          })}
+        </tbody>
+      </table>
+      <div className={tableStyles.containterButton}>
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.js:53]: The tablet should not contain the buttons to redirect to the form.

File: src/Components/Shared/Table/index.module.css [hunk 14]
```diff
@@ -0,0 +1,31 @@
+.buttonAdd{
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.module.css:1]:
```suggestion
.buttonAdd {
```
Review the spaces here. We should left an space before the `{` and we have to add a break line between each class.

File: src/Components/Shared/Table/index.module.css [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+    border-radius: 5px;
+    border: 1px solid black;
+    font-size: 16px;
+    cursor: pointer;
+}
+.buttonAdd:hover{
+    border: 2px solid #EA3354;
+    box-shadow: 0px 0px 8px #EA3354;
+    background-color: rgb(202, 200, 200);
+
```
Comment (lmalvasia) [inline src/Components/Shared/Table/index.module.css:13]: Remove emptyy spaces

File: src/Components/Admins/Modal/modal.module.css (hunk 2 above)
Comment (lmalvasia) [inline src/Components/Admins/Modal/modal.module.css:1]: I think we could remove this file.

File: src/Components/Employees/index.js [hunk 16]
```diff
@@ -50,17 +49,15 @@ const Employees = () => {
       </Modal>
       <h2>Employees</h2>
       <div>
-        <ListEmployees
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:53]: Looks like you are not rendering the list employees, so could we remove this another component?

File: src/Components/SuperAdmins/index.js [hunk 17]
```diff
@@ -33,13 +35,13 @@ const SuperAdmins = () => {
   };
 
   return (
-    <div>
-      <List
```
Comment (lmalvasia) [inline src/Components/SuperAdmins/index.js:37]: If we are not using this, could we remove it? List and list item

File: src/Components/TimeSheets/index.js [hunk 18]
```diff
@@ -1,32 +1,57 @@
...
     return new Date(date).toISOString().split('T')[0];
   };
+
+  const timeSheetData = () => {
+    return timesheets.map((timesheet) => {
+      return {
+        ...timesheet,
+        date: dateFormatted(timesheet.date),
+        task: !timesheet.task ? (
+          <p className={styles.none}>There is no task</p>
```
Comment (lmalvasia) [inline src/Components/TimeSheets/index.js:30]:
```suggestion
          There is no task
```

Comment (lmalvasia) [inline src/Components/TimeSheets/index.js:30]: Why are we using a `<p>` here?

File: src/Components/Shared/Table/index.js (hunk 10 above)
Comment (EmanuelMuises95) [reply src/Components/Shared/Table/index.js]: Done

Comment (EmanuelMuises95) [reply src/Components/Shared/Table/index.js]: Changed

File: src/Components/Shared/Table/index.js (hunk 11 above)
Comment (EmanuelMuises95) [reply src/Components/Shared/Table/index.js]: Replaced

File: src/Components/Shared/Table/index.js (hunk 12 above)
Comment (EmanuelMuises95) [reply src/Components/Shared/Table/index.js:37]: Change done

File: src/Components/TimeSheets/index.js (hunk 18 above)
Comment (EmanuelMuises95) [reply src/Components/TimeSheets/index.js:31]: Sorry, this was a conditional so that if it was empty it would set a default message

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**